CMakeLists.txt

```diff
@@ -442,6 +443,26 @@ set (UPM_COMMON_HEADER_DIRS ${CMAKE_HOME_DIRECTORY}/include)
 # Generate a build-only C++ header to add functionality to SWIG'ed modules
 configure_file (${PROJECT_SOURCE_DIR}/cmake/modules/version.hpp.in ${PROJECT_BINARY_DIR}/src/version.hpp @ONLY)
 
+# Cordova binding
+if (BUILDCORDOVABINDING)
+  if (BUILDSWIGJAVA)
```

Comment:
you could use `if (NOT BUILDSWIGJAVA)` instead of using the else body for this.

Also, you could just wrap the entire body of this with a `if (BUILDSWIGJAVA)` because cordova bindings only make sense if the Java bindings are enabled

Reply:
Thanks! But I still think it's better to put the body of this outside of `BUILDSWIGJAVA` to let everyone who want to enable Cordova binding realize that it's based on Java modules.

Reply:
Make sense. Then we should improve the error message if BUILDSWIGJAVA is not enabled! The reason behind that is the amount of stuff being executed with cmake (in travis for example). If something fails, I want to know specifically what stuff failed, so the error message is something really important to quickly identify the root cause here